### PR TITLE
updates roadmap for sept 2022

### PR DESCRIPTION
**Pull request by jpvajda**
Sept 2022 roadmap updates.

### CODE REVIEW

**Comment by jpvajda** [comment]
@martinbonnin 👋  let me know how this sounds to you. 😎

File: ROADMAP.md [hunk 1]
```diff
@@ -15,9 +15,9 @@ For up to date release notes, refer to the project [Change Log](https://github.c
 
 ## Upcoming Releases
 
-### 3.6
+### 3.7
 
-- Release 3.6 will focus on adding support for the `@defer` directive as well as adding `schemaBased` test builders. See Github [3.6 Milestone](https://github.com/apollographql/apollo-kotlin/milestone/23) for more details.
+- Release 3.7 will focus on providing better Java support in Apollo Kotlin.  See Github [3.7 Milestone](https://github.com/apollographql/apollo-kotlin/milestone/25) for more details.
```
Comment (martinbonnin) [inline ROADMAP.md:20]: I'd remove the specific `3.7` version here. There's a high chance we'll need to unbundle the Java runtime from the Kotlin one and therefore version numbers will not align

**Comment by martinbonnin** [comment]
We can remove `Test APIs` from the longer term section. This is mostly done with `data builders`

File: ROADMAP.md (hunk 1 above)
Comment (jpvajda) [reply ROADMAP.md:20]: So how would we communicate this release? Do you think it's going to a new version starting at 1.0 for Java?

Comment (jpvajda) [reply ROADMAP.md]: great. I left it as release `TBD` and we can decide the order of the releases as we break this work down further.

**Pull request merged**